docs/DeveloperGuide.md

```diff
@@ -1,34 +1,216 @@
 # Developer Guide
+- [Design](#1-Design)
```

Comment:
Link does not work. Perhaps this is caused by not using lowercase for the id of the link. Also, not every section is listed in the table of contents but I think this is because the guide is still incomplete.